### PR TITLE
dev/core#980 [phpunit test] update phpunit extends class to support later versions

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Updates phpunit version support in the test suite

Before
----------------------------------------
Supported versions of phpunit (7 & 8) do not work with CiviCRM test suite.

After
----------------------------------------
Supported versions of phpunit (7 & 8) do work with CiviCRM test suite. Versions prior to 4.8 do not

Technical Details
----------------------------------------
Per https://engineering.facile.it/blog/eng/phpunit-upgrade-namespace/ this works with phpunit 4.8 +

Phpunit support for php versions is here https://phpunit.de/supported-versions.html

Basically we need phpunit 7 to support php 7.3 - which we should be adding to our
test suite now. It's hard to make a case that we still need to support phpunit 4.8 - let
alone versions before that - 5.7 still covers php 5.6 which we need for some of our current branches.

Unit testing on master would ideally switch up to a supported version of phpunit - 7 or 8

Comments
----------------------------------------
@totten @seamuslee001

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by seamuslee001** [comment]
Given this only affects tests and Jenkins is ok with it then i think we should merge it, Jenkins should be testing primarily using phpunit5. Given that min is 5.6 it should be caught by this code

```
if (version_compare(PHP_VERSION, '5.6', '>=')) {
  $phpunit = findCommand('phpunit5');
}
```

that is in `tools/scripts/phpunit`

**Comment by eileenmcnaughton** [comment]
woot

**Comment by totten** [comment]
Sweet. Agree this is a good idea for forward compatibility. And in this case... if it passes (esp on `bknix-{min,max,dfl}`), then it passes.  :)

**Comment by seamuslee001** [comment]
@totten we are hitting some issues with further changes that are needed for a PHPUnit6 test runner see #14387
